workers/unity/Packages/io.improbable.gdk.buildsystem/Configuration/BuildConfigEditor.cs

```diff
@@ -632,28 +633,56 @@ private GUIContent GetBuildTargetGuiContents(BuildTargetConfig c)
 
         private void DrawBuildTargets(BuildEnvironmentConfig env, int hash)
         {
+            // Build a mapping the build target position in the toolbar and it's position in the BuildTargets array.
+            // Sometimes there is a different of targets in the BuildTargets array and those in the toolbar as
+            // some are deprecated
+            if (toolbarMapping.Count == 0)
+            {
+                var index = 0;
+                var skipped = 0;
+                env.BuildTargets.ForEach(t =>
+                {
+                    if (t.Deprecated)
+                    {
+                        skipped++;
+                    }
+                    else
+                    {
+                        toolbarMapping.Add(index++, skipped++);
+                    }
+                });
+            }
+
             // Init cached UI state.
             var selectedBuildTarget = stateManager.GetStateObjectOrDefault<BuildTargetState>(hash ^ typeof(BuildTargetState).GetHashCode());
 
             if (selectedBuildTarget.Choices == null || invalidateCachedContent > 0)
             {
-                selectedBuildTarget.Choices = env.BuildTargets.Select(GetBuildTargetGuiContents).ToArray();
+                selectedBuildTarget.Choices = env.BuildTargets
+                    .Where(c => !c.Deprecated)
+                    .Select(GetBuildTargetGuiContents)
+                    .ToArray();
             }
 
+
             // Draw available build targets.
             using (new EditorGUIUtility.IconSizeScope(SmallIconSize))
             {
                 selectedBuildTarget.Index =
                     GUILayout.Toolbar(selectedBuildTarget.Index, selectedBuildTarget.Choices);
             }
 
+            // Convert the index from the position selected in the toolbar to the position in the BuildTargets array
+            var selectedTargetIndex = toolbarMapping[selectedBuildTarget.Index];
+
             // Draw selected build target.
-            var buildTarget = env.BuildTargets[selectedBuildTarget.Index];
+            var buildTarget = env.BuildTargets[selectedTargetIndex];
             var canBuildTarget = BuildSupportChecker.CanBuildTarget(buildTarget.Target);
 
             var options = buildTarget.Options;
             var enabled = buildTarget.Enabled;
             var required = buildTarget.Required;
+            var deprecated = buildTarget.Deprecated;
```

Comment:
Isn't this impossible to be `true` here?

Reply:
Yes, I think this was a remnant of a previous attempt to get it working, so it can be removed safely